webapp/channels/src/components/emoji/emoji_list_item/emoji_list_item.tsx

```diff
@@ -22,72 +22,71 @@ export type Props = {
     actions: {
         deleteCustomEmoji: (emojiId: string) => void;
     };
-}
+};
 
-export default class EmojiListItem extends React.PureComponent<Props> {
-    static defaultProps = {
-        emoji: {} as CustomEmoji,
-        currentUserId: '',
-        creatorDisplayName: '',
-    };
+const DELETE_PERMISSION = [Permissions.DELETE_EMOJIS];
+const DELETE_OTHER_PERMISSION = [Permissions.DELETE_OTHERS_EMOJIS];
 
-    handleDelete = (): void => {
-        if (!this.props.emoji) {
+const EmojiListItem = ({
+    actions: {
+        deleteCustomEmoji,
+    },
+    onDelete,
+    emoji = {} as CustomEmoji,
+    creatorUsername,
+    currentUserId = '',
+    creatorDisplayName = '',
+}: Props) => {
+    let displayName = creatorDisplayName;
+    const handleDelete = useCallback((): void => {
```

Comment:
Can you bring the use memo up here? and move the displayName a bit down? I like components to have the following structure:
- Simple assign to variable hooks (useMemo, useState, useRef...)
- Callbacks
- Effects
- Rendering logic

`displayName` is part of the rendering logic, so having it before the callback is confusing, and I like all memos to be together in the same place, to avoid also confusing ordering.

Reply:
Oh yes, makes sense.
Will make sure to follow this everywhere else as well. 🙌